src/Tools/VMC/EventLibraryInterface.cxx

```diff
@@ -0,0 +1,261 @@
+
+  unsigned int nLep = 0;
+  for(const Particle& part: rec->parts) if(pdg::IsLepton(part.pdg)) ++nLep;
+
+  const int firstLep = (nLep == 0) ? -1 : 2;
+  const int lastLep  = (nLep == 0) ? -1 : 1+nLep;
+
+  const int firstHad = (rec->parts.size() == nLep) ? -1 : 2+nLep;
+  const int lastHad  = (rec->parts.size() == nLep) ? -1 : 1+rec->parts.size();
+
+  // Neutrino is a parent to the lepton(s)
+  event->AddParticle(init_state.ProbePdg(),
+                     kIStInitialState,
+                     -1, -1, firstLep, lastLep,
+                     *probe_p4,
+                     TLorentzVector(0, 0, 0, 0));
+
+  const int tgt_A    = init_state.Tgt().A();
+  const int tgt_Z    = init_state.Tgt().Z();
+  const int tgt_pdgc = pdg::IonPdgCode(tgt_A, tgt_Z);
+  // REVIEW: it was like this in Costas' example code. Can it just be
+  // init_state.TgtPdg()?
+
+  LOG("ELI", pINFO)
+    << "Adding nucleus [A = " << tgt_A << ", Z = " << tgt_Z
+    << ", pdg = " << tgt_pdgc << "]";
+
+  // Nucleus is a parent to everything else
+  event->AddParticle(tgt_pdgc,
+                     kIStInitialState,
+                     -1, -1, firstHad, lastHad,
+                     0, 0, 0, PDGLibrary::Instance()->Find(tgt_pdgc)->Mass(),
+                     0, 0, 0, 0);
+
+  const std::vector<TVector3> basis = Basis(probe_p4->Vect());
+
+  for(bool lep: {true, false}){ // make sure lepton in first
+    for(const Particle& part: rec->parts){
+      if(lep != pdg::IsLepton(part.pdg)) continue;
+
+      // Fix up the outgoing lepton for NC events (in the library it's always
+      // nu_mu...)
+      int pdg = part.pdg;
+      if(interaction->ProcInfo().IsWeakNC()) pdg = init_state.ProbePdg();
+
+      event->AddParticle(pdg,
+                         kIStStableFinalState,
+                         (lep ? 0 : 1), -1, -1, -1, // child of the neutrino or nucleus
+                         TLorentzVector(part.px*basis[0] +
+                                        part.py*basis[1] +
+                                        part.pz*basis[2],
+                                        part.E),
+                         TLorentzVector(0, 0, 0, 0));
+    }
+  }
+}
+
+//____________________________________________________________________________
+const Record* EventLibraryInterface::GetRecord(const Interaction* interaction) const
+{
+  if(fRecords.empty()) LoadRecords();
+
+  const InitialState& init_state = interaction->InitState();
+
+  const double probe_E = init_state.ProbeE(kRfLab);
+
+  if(!init_state.Tgt().IsNucleus()){
+    LOG("ELI", pINFO) << "Skippping non-nuclear target " << init_state;
+    return 0;
+  }
+
+  const int tgt_pdgc = init_state.TgtPdg();
+
+  const ProcessInfo& proc = interaction->ProcInfo();
+
+  if(!proc.IsWeakCC() && !proc.IsWeakNC()){
+    LOG("ELI", pINFO) << "Skipping unknown process " << proc;
+    return 0;
+  }
+
+  int probe_pdgc = init_state.ProbePdg();
+
+  // Use nu_mu for NC by convention
+  if(proc.IsWeakNC()){
```

Comment:
This convention of tagging all the NC's a nu_mu generated is something that one has to impose when generating the event library, then?

Reply:
The library has directories like cc/nu_mu, but it also has nc/nu and nc/nu_bar. This NC=numu convention is internal to the code. I also generated them that way, but it doesn't matter. I rewrite the outgoing neutrino flavour elsewhere.

Reply:
Improved the comments.